src/api/internal/bam/BamWriter_p.h

```diff
@@ -45,7 +46,8 @@ class BamWriterPrivate {
         bool IsOpen(void) const;
         bool Open(const std::string& filename,
                   const std::string& samHeaderText,
-                  const BamTools::RefVector& referenceSequences);
+                  const BamTools::RefVector& referenceSequences,
+				  IBamIODevice* device = 0);
```

Comment:
You should probably fix the formatting here.  Also it looks like you should be using `NULL` in this repository, not `0`.